doc/subsystems/test/ztest.rst

```diff
@@ -101,6 +101,138 @@ and are used to decide whether a test failed or passed by verifying whether an
 interaction with an object occurred, and if required, to assert the order of
 that interaction.
 
+.. _main_c_bp:
+
+Best practices for declaring the test suite
+===========================================
+
+*sanitycheck* and other validation tools need to obtain the list of
+subcases that a Zephyr *ztest* test image will expose.
+
+.. admonition:: Rationale
+
+   This all is for the purpose of traceability. It's not enough to
+   have only a semaphore test project.  We also need to show that we
+   have testpoints for all APIs and functionality, and we trace back
+   to documentation of the API, and functional requirements.
+
+   The idea is that test reports show results for every sub-testcase
+   as passed, failed, blocked, or skipped.  Reporting on only the
+   high-level test project level, particularly when tests doing too
+   many things, is too vague.  Generic template
```

Comment:
Start a new paragraph:

    Here is a generic template for a test showing the
    expected use of :func:`ztest_test_suite`:

Reply:
done, with normal indentation--the template it should not be part of the rationale